agcaballero/portfolio/src/main/webapp/script.js

```diff
@@ -65,3 +65,44 @@ function testRandomPost(trials = 10000) {
 }
 
 testRandomPost(10000);  // runs 10000 trials on randomPost()
+
+/** Checks validity the information from the "Contact Me" form */
+function isValid() {
+  // here are fields that need more validation than just checking if not empty
+  const email = document.getElementById(email);
+
+  const fields = [
+    document.getElementById('first-name'),
+    document.getElementById('last-name'),
+    email,
+    document.getElementById('subject'),
+    document.getElementById('body'),
+  ];
+
+  // checks if any field is empty, if so returns false
+  for (let i = 0; i < fields.length; i++) {
+    if (isNotEmpty(fields[i].value)) continue;
+
+    return false;
+  }
+
+  return isEmail(email.value);
+}
+
+/** Return true if value is not null or undefined, false if they are */
+function isNotEmpty(value) {
+  return Boolean(value);  // if null or undefined, will return false
```

Comment:
The style guide prohibits wrapping primitive values:
https://google.github.io/styleguide/jsguide.html#disallowed-features-wrapper-objects

It's unclear whether `value` is primitive or not... is there a different way to check this?

Note also that the documentation indicates that a primitive `boolean` is returned, while a `Boolean` object is actually return, which could be a source of confusion. Consider implementing like:

```
if (value == null) {
  return true;
} else if (...) {
  return ...;
}
return ...;
```

Reply:
I changed it so it returns true or false based on if-else as suggested, so there should now be no problem with wrapping a primitive.

Reply:
Done